include/beman/inplace_vector/inplace_vector.hpp

```diff
@@ -42,13 +43,42 @@ using inplace_vector_internal_size_type =
           If<Capacity <= std::numeric_limits<uint32_t>::max(), uint32_t,
              uint64_t>>>;
 
+// array based storage is used so that we can satisfy constexpr requirement
+//
+// Selecting this storage type implies: std::is_trivial_v<T>
+template <typename T, std::size_t Capacity>
+struct inplace_vector_array_based_storage {
+  std::array<T, Capacity> elems;
+
+  constexpr T *begin() { return elems.data(); }
+  constexpr const T *begin() const { return elems.data(); }
+};
+
+// byte array based storage is used for non-constexpr environment, where default
+// initialization may not be available.
+//
+// Selecting this storage type implies: !std::is_trivial_v<T>
+template <typename T, std::size_t Capacity>
+struct inplace_vector_bytes_based_storage {
+  alignas(T) unsigned char elems[Capacity * sizeof(T)];
```

Comment:
- `std::byte` instead of `unsigned char`?
- compile error on some compilers (e.g. MSVC) if `Capacity == 0`

Reply:
> I think this is o.k. Why would we need initialisation before actually having any elements in the vector?

Does default constructor of std::array default construct all elements? I thought declaring uninitialized value is not allowed (including aggregate).

> std::byte instead of unsigned char?

You are right, probably should use std::byte.

> compile error on some compilers (e.g. MSVC) if Capacity == 0

Agh! I think we might need a template specialization for `Capacity = 0`, I don't have MSVC locally, do you mind to sharing the error message?

Reply:
Okay I added the const to std::array, I will investigate non-constexpr version once I have the test cases setup.

Reply:
Switched to std::byte.